crates/spirv-std/src/textures.rs

```diff
@@ -61,36 +61,36 @@ pub struct StorageImage2d {
 impl StorageImage2d {
     /// Read a texel from an image without a sampler.
     #[spirv_std_macros::gpu_only]
-    pub fn read<I, V, V2>(&self, coordinate: V) -> V2
+    #[cfg(feature = "const-generics")]
+    pub unsafe fn read<I, V, const N: usize>(&self, coordinate: impl Vector<I, 2>) -> V
```

Comment:
I'm assuming there was some discussion somewhere on why `read` should be unsafe - could you add context of why it's unsafe in doc comments? (rather than just `write` being unsafe, which IIRC is what we agreed on last week)

Reply:
Hmm, no I don't think we did that, and looking at this now I'm not sure why I changed that, removing.
```suggestion
    pub fn read<I, V, const N: usize>(&self, coordinate: impl Vector<I, 2>) -> V
```